### ISSUE TITLE
OS X: borg mount files are not accessible by finder

### ISSUE THREAD

**Issue by tristan-k** [borgbackup/borg#1126]
Mounting the borg repo works fine but the actual files stored inside are not accessible. In the mac os x finder (file browser) they all have a size of 0 Byte. Trying to copy one file outside the fuse volume results in a `Error -36`.

```
$ borg --version
borg 1.0.4.dev273+ngbf5406d
$ borg mount --foreground ~/Desktop/borg-repo ~/Desktop/BorgBackup
```

***
The bounty is for finding the root cause of this issue and showing a way to fix it.
A PR with a fix would be welcome (if major work, bounty could be increased / new bounty filed for the code).

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/34843144-os-x-borg-mount-files-are-not-accessible-by-finder)

**Comment by ThomasWaldmann** [comment]
Some minutes ago I tried `borg mount` on OS X with osxfuse 3.2.0 and it worked. File sizes correct, files readable.

So, did they do incompatible changes? Or introduce a regression?

**Comment by tristan-k** [comment]
This must be related to the `Finder`. I did browse into the mounted repo with a terminal and the files are accessible there.

**Comment by ThomasWaldmann** [comment]
Maybe there should be a ticket about this in the osxfuse tracker - can you file one there?

**Comment by tristan-k** [comment]
I've just opened a ticket at https://github.com/osxfuse/osxfuse/issues/288

**Comment by tristan-k** [comment]
> I think this is something the Borg team needs to investigate. The loopback reference file system works fine with osxfuse 3.3.2. Finder acting up can mean that the root user does not have access to the volume.

[Quote](https://github.com/osxfuse/osxfuse/issues/288#issuecomment-223642008)

**Comment by ThomasWaldmann** [comment]
@tristan-k can you try whether 3.2.0 works with finder?

**Comment by tristan-k** [comment]
@ThomasWaldmann No it did not. Tried that before upgrading to latest version.

**Comment by ThomasWaldmann** [comment]
OK, if no borg + osxfuse version works with finder (but with shell), update issue topic and top text accordingly.

**Comment by ThomasWaldmann** [comment]
The hint about root user helped, i can reproduce on the shell:
```
(borg-env) This-MacBook-Pro:borg vagrant$ ls /vagrant/borg/mnt
borg
(borg-env) This-MacBook-Pro:borg vagrant$ sudo ls /vagrant/borg/mnt
ls: mnt: No such file or directory
```

**Comment by ThomasWaldmann** [comment]
Solved!:
```
(borg-env) This-MacBook-Pro:borg vagrant$ ./borg.exe mount repo::1 mnt -o allow_other
(borg-env) This-MacBook-Pro:borg vagrant$ ls /vagrant/borg/mnt
borg
(borg-env) This-MacBook-Pro:borg vagrant$ sudo ls /vagrant/borg/mnt
borg
```

**Comment by ThomasWaldmann** [comment]
Can you try this with finder?

Guess we should put a hint about this option into the docs.

**Comment by tristan-k** [comment]
Just did. Doesn't work in the finder.

**Comment by ThomasWaldmann** [comment]
@tristan-k are you in "admin" group (this is needed, otherwise "allow_other" is not allowed for you).

**Comment by enkore** [comment]
Q: What produces this? Default ACLs? Can't be unix permissions for sure.

**Comment by tristan-k** [comment]
Yes, I'am in the admin group.

**Comment by ThomasWaldmann** [comment]
What does error -36 (see top post) mean in that context?

**Comment by ThomasWaldmann** [comment]
Help wanted: we can only fix this, if we know the root cause (and how to fix it):

Why is finder not seeing files in the same was as `ls -l` on the shell?

**Comment by tristan-k** [comment]
@ThomasWaldmann I can give you remote access to my mac os x system if you like.

**Comment by ThomasWaldmann** [comment]
@tristan-k OK, let's meet on #borgbackup on chat.freenode.net?

**Comment by ThomasWaldmann** [comment]
I am upgrading Vagrantfile to use osxfuse 3.3.3 now.

From the changelog:
```
    Fix invisible volumes on OS X 10.8 and earlier

    Wait for FUSE volume to be initialized instead of faking vnop_getattr for the volume's root vnode. Faking attributes results in Finder not showing the mounted volume on OS X 10.8 and earlier.
```

Does that solve this ticket?

**Comment by ThomasWaldmann** [comment]
fixed in 1.0-maint
